src/components/solvers/ProgressHandler.tsx

```diff
@@ -1,34 +1,48 @@
-import React, { useState } from "react";
-import { GoButton } from "./buttons/GoButton";
-import { postProblem } from "../../api/ToolboxAPI";
+import React, {useState} from "react";
+import {GoButton} from "./buttons/GoButton";
+import {postProblem} from "../../api/ToolboxAPI";
 import {ProgressView} from "./ProgressView";
-import { Container } from "../Container";
+import {Container} from "../Container";
 import {Solution} from "./Solution";
+import {SolverPicker} from "./SolverPicker";
+import {ProblemSolver} from "./ProblemSolver";
 
 export interface ProgressHandlerProps {
     problemType: string;
     problemInput: any;
 }
 
 export const ProgressHandler = (props: ProgressHandlerProps) => {
-    const [wasClicked, setClicked] = useState(false);
-    const [finished, setFinished] = useState(false);
+    const [wasClicked, setClicked] = useState<boolean>(false);
+    const [solverPicked, setSolverPicked] = useState<boolean>(false);
+    const [finished, setFinished] = useState<boolean>(false);
     const [solution, setSolution] = useState<Solution>();
 
     async function onGoClicked() {
         setClicked(true);
+        setSolverPicked(false);
         setFinished(false);
-        let solution = await postProblem(props.problemType, props.problemInput);
-        setSolution(solution);
-        setFinished(true);
+    }
+
+    async function startSolving(solver: ProblemSolver | null) {
+        setSolverPicked(true);
+
+        postProblem(props.problemType, props.problemInput, solver)
+            .then(solution => {
+                setSolution(solution);
+                setFinished(true);
+            });
     }
 
     return (
         <Container>
             {!wasClicked || finished
                 ? <GoButton clicked={onGoClicked}/>
                 : null}
-            {wasClicked
+            {wasClicked && !solverPicked
+                ? <SolverPicker problemType={props.problemType} onSolverPicked={startSolving}/>
+                : null}
+            {wasClicked && solverPicked
```

Comment:
I don't think that would work well when we want to distinguish between selecting the Meta Solver and specific solvers.
Everything would be in one <Select> group and we wouldn't show the Meta Solver as prominently.
I'd say let's wait for @koalamitice's feedback.

Reply:
Good point but I think we can raise attention to the meta solver either by using `<optgroup>`s (maybe "automated" and "manual"?) or by making the meta-solver option bold. I think we should select the meta-solver by default anyways (and iirc the backend does this already)

Reply:
I can also make a quick test version for that on another branch.
Another issue with putting the selection directly next to the Go Button might be multi step solvers. We don't really have a concept for that yet, but Domenik told us that the solvers would be selected step by step. I wouldn't put a lot of selections next to the go button in this case.
That might not be as relevant for single problem masks like SAT or MaxCut, but I'd still keep it as a separate component for this just in case. This approach would also still compatible with the single `<Select>` idea which includes the Meta Solver.

Reply:
Alternative version here:
https://github.com/ProvideQ/ProvideQ/tree/solver-picker-v2

I think I like it more. Sadly, the <option> doesn't give us a lot of option for custom styling. We can show both to Domenik can pick what he likes best.

Reply:
> Another issue with putting the selection directly next to the Go Button might be multi step solvers. We don't really have a concept for that yet, but Domenik told us that the solvers would be selected step by step. I wouldn't put a lot of selections next to the go button in this case.

Agreed, our UI would look pretty cluttered. I'm not sure if multi-step-solvers would be available in the UI anyways so let's discuss that in our next meeting.

> Alternative version here: https://github.com/ProvideQ/ProvideQ/tree/solver-picker-v2
> 
> I think I like it more. Sadly, the doesn't give us a lot of option for custom styling. We can show both to Domenik can pick what he likes best.

I like this one better too but sure, let's ask Domenik. One more thing: I think we can remove the text above the selector and use a tooltip instead.